tests/monitoring/metrics.go

```diff
@@ -0,0 +1,139 @@
+			"kubevirt_vmi_migration_succeeded":                                   true,
+			"kubevirt_vmi_migration_failed":                                      true,
+		}
+
+		It("should contain virt components metrics", func() {
+			metrics = fetchPrometheusMetrics(virtClient)
+
+			err = virtoperator.SetupMetrics()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = virtapi.SetupMetrics()
+			Expect(err).ToNot(HaveOccurred())
+
+			err = virtcontroller.SetupMetrics(nil, nil, nil, nil, nil, nil, nil, nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, metric := range operatormetrics.ListMetrics() {
+				if excludedMetrics[metric.GetOpts().Name] {
+					continue
+				}
+
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, nil)))
+			}
+		})
+
+		It("should have kubevirt_vmi_phase_transition_time_seconds buckets correctly configured", func() {
+			buckets := virtcontroller.PhaseTransitionTimeBuckets()
+
+			for _, bucket := range buckets {
+				labels := map[string]string{"le": strconv.FormatFloat(bucket, 'f', -1, 64)}
+
+				metric := operatormetrics.NewHistogram(
+					operatormetrics.MetricOpts{Name: "kubevirt_vmi_phase_transition_time_from_deletion_seconds"},
+					prometheus.HistogramOpts{},
+				)
+
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, labels)))
+			}
+		})
+	})
+})
+
+func fetchPrometheusMetrics(virtClient kubecli.KubevirtClient) *libmonitoring.QueryRequestResult {
+	metrics, err := libmonitoring.QueryRange(virtClient, "{__name__=~\"kubevirt_.*\"}", time.Now().Add(-1*time.Minute), time.Now(), 15*time.Second)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(metrics.Status).To(Equal("success"))
+	Expect(metrics.Data.ResultType).To(Equal("matrix"))
+	Expect(metrics.Data.Result).ToNot(BeEmpty(), "No metrics found")
+
+	return metrics
+}
+
+func setupVM(virtClient kubecli.KubevirtClient) {
+	vm := createRunningVM(virtClient)
+	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", 1)
+
+	By("Deleting the VirtualMachine")
+	err := virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", -1)
+}
+
+func createRunningVM(virtClient kubecli.KubevirtClient) *v1.VirtualMachine {
+	vmi := libvmifact.NewGuestless(libvmi.WithNamespace(testsuite.GetTestNamespace(nil)))
+	vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunning())
+	vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm)
+	Expect(err).ToNot(HaveOccurred())
+
+	Eventually(func() bool {
+		vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return vm.Status.Ready
+	}, 300*time.Second, 1*time.Second).Should(BeTrue())
```

Comment:
Isn't the `libwait.WaitForSuccessfulVMIStart(vmi)` enough?

Reply:
and perform a vm.Get after it?

Reply:
The VMI still takes a few seconds to be created after you create the VM. And `WaitForSuccessfulVMIStart` just waits for the status, but if it does not find the VMI it exists with an error